.github/workflows/ci.yml

```diff
@@ -1,48 +1,118 @@
 name: CI
-
-on:
+'on':
   push:
     branches:
       - main
   pull_request:
     branches:
       - main
-
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  group: '${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}'
   cancel-in-progress: true
-
 jobs:
   check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-
-      - name: Install deps
-        run: sudo apt-get install libfmt-dev libgflags-dev
-
       - name: Format
         run: cargo fmt --all -- --check
-
+      
       - name: Clippy
         working-directory: cxx-async
         run: cargo clippy -- -D warnings
-
+  
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: '${{ matrix.os }}'
     strategy:
       matrix:
         os:
           - ubuntu-latest
-    #          - macos-latest
-    #          - windows-latest
+    
     steps:
       - uses: actions/checkout@v4
-
-      - name: Install dependencies on linux
-        if: matrix.os == 'ubuntu-latest'
-        run: sudo apt-get install libfmt-dev libgflags-dev
-
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            git \
+            cmake \
+            make \
+            gcc \
+            g++ \
+            clang \
+            autoconf \
+            automake \
+            libtool \
+            python3-pip \
+            ninja-build \
+            libgflags-dev \
+            libdouble-conversion-dev \
+            libevent-dev \
+            libgflags-dev \
+            libgoogle-glog-dev \
+            libssl-dev \
+            libunwind-dev \
+            liblz4-dev \
+            liblzma-dev \
+            zlib1g-dev \
+            libsnappy-dev \
+            binutils-dev \
+            libjemalloc-dev \
+            libiberty-dev 
+      
+      - name: Build fmt
+        working-directory: /tmp/
+        run: |
+          git clone https://github.com/fmtlib/fmt.git && cd fmt
+          rm _build -rf; mkdir _build -p && cd _build
+          cmake .. -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
+          -DCMAKE_C_COMPILER=/usr/bin/clang \
+          -DFMT_TEST=OFF
+          make -j$(nproc)
+          sudo make install
+          sudo ldconfig
+      
+      - name: Build fast_float
+        working-directory: /tmp/
+        run: |
+          git clone https://github.com/fastfloat/fast_float.git && cd fast_float
+          rm _build -rf; mkdir _build -p && cd _build
+          cmake .. -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
+          -DCMAKE_C_COMPILER=/usr/bin/clang \
+          -DCMAKE_CXX_FLAGS="-O3" \
+          -DBUILD_SHARED_LIBS=TRUE \
+          -DBUILD_BENCHMARKS=OFF
+          make -j$(nproc)
+          sudo make install
+          sudo ldconfig
+      
+      - name: Build Folly
```

Comment:
Nice and thank you! We can cache them so that we don't need to build them every time. 

Please take a look over opendal's logic for start: https://github.com/apache/opendal/blob/fab80c9601a8fd128176586e92a31d09f75062fb/.github/actions/setup/action.yaml#L104-L130